src/section/section_proof_chain.rs

```diff
@@ -6,114 +6,85 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use crate::{
-    error::Result,
-    xor_space::{Prefix, XorName},
-};
-
-#[cfg(all(test, feature = "mock"))]
-use std::iter;
+use std::{collections::HashSet, iter};
 
+/// Block of the section proof chain. Contains the section BLS public key and is signed by the
```

Comment:
who will sign the first block?
or this block will only be in `tail`, not in `head`?
in that case, maybe the comment better to mention this explicitly?

Reply:
The first block is not signed. If this chain is the full section chain, then the first block is the genesis block created by the very first node in the network. If this is just a slice of a chain, the presumably the first block's key is already trusted (by being verified previously) so it doesn't need to be signed.

> or this block will only be in tail, not in head?

That's right. The first block doesn't have signature so it's not stored as `SectionProofBlock` (which is `key` + `signature`) but only as the key alone.

Reply:
In this case, I'd prefer to make the comment to be more explicit, say:
Block of the section proof chain (tail part only)